CDTDatastore/HTTP/CDTRequestLimitInterceptor.m

```diff
@@ -50,16 +49,32 @@ - (instancetype)initWithSleep:(NSTimeInterval)sleep
  */
 - (CDTHTTPInterceptorContext *)interceptResponseInContext:(CDTHTTPInterceptorContext *)context
 {
-    if (context.response.statusCode == 429 && self.maxRequests > 0) {
+    if (context.response.statusCode == 429) {
+
+        // if we are the first invocation in this pipeline, set some state
+        if (!context.state[@"sleep"]) {
```

Comment:
We should prefix the key for state so it looks something like: `CDTRequestLimitInterceptor.xxx`